omod/src/main/java/org/openmrs/module/owa/web/controller/OwaRestController.java

```diff
@@ -151,77 +157,90 @@
 
 	@RequestMapping(value = "/rest/owa/installapp", method = RequestMethod.POST)
 	@ResponseBody
-	public List<App> install(@RequestBody InstallAppRequestObject urlObject, HttpServletRequest request, HttpServletResponse response) throws IOException {
+	public ResponseEntity<List<App>> install(@RequestBody InstallAppRequestObject urlObject,
+	        HttpServletRequest request, HttpServletResponse response) throws IOException {
+		String message;
+		String fileName = null;
 		List<App> appList = new ArrayList<>();
+		Boolean checkInstall = false;
 
-		String url = urlObject.getUrlValue();
-		String fileName = urlObject.getFileName();
-		if(fileName == null || fileName == ""){
-			throw new  NullPointerException(
-				"File name must be passed for installation"
-			);
-		}
-		URL downloadUrl = null;
-		if (ResourceUtils.isUrl(url)) {
-			downloadUrl = new URL(url);
-		}
 		if (Context.hasPrivilege("Manage OWA")) {
-			String message;
 			HttpSession session = request.getSession();
-			if (!url.isEmpty()) {
+			URL downloadUrl = null;
+			String installUrl = urlObject.getUrlValue();
+
+			if (!installUrl.contains(".zip")) {
+				message = messageSourceService.getMessage("owa.invalid_owa_Url");
+				log.warn("Invalid URL to OWA download");
```

Comment:
The log warning would be better if it tells the expected file format

Reply:
I've modified the log warning